### PR TITLE
[Outreachy Task Submission] Added a cancel button on the general page in settings which has a capability of confirming to user if they want to cancel the action

### PR DESCRIPTION
**Pull request by kivuvarosekivuvan**
# Engineers Checklist

## What this PR does?
- Introduced a 'Cancel' button on the general settings page to allow users to discard changes without saving them.
- Added pop up to confirm to the user if they want to cancel the action.
- Introduced a confirmation message when successfully canceled.
- Ensures that if a user doesn't interact with the form, they are unable to discard the existing values.
- In such cases, a pop-up message informs them that no changes have been made yet. 

## How can it be tested?
- Launch the local server: http://localhost:4200
- Login/signup to have the settings feature
- Navigate to the general page in settings
- Scroll to the bottom,click on the 'cancel' button

# Where is it documented?
https://github.com/ushahidi/platform/issues/4737

## What would be the Impact of the solution?
- The introduction of a "Cancel" button on the general settings page, along with a pop-up confirmation and a success message, will improve the user experience by providing a convenient way to discard changes, preventing unintended modifications, and enhancing user control and efficiency

### The button added
Screenshot from 2024-03-21 17-16-59

### Confirmation dialog
Screenshot from 2024-03-11 18-51-43
b7287a)

### The confirmation message once "OK" button is clicked

Screenshot from 2024-03-21 17-14-11

## Video walkthrough

[Screencast from 03-21-2024 05:13:13 PM.webm](https://github.com/ushahidi/platform-client-mzima/assets/124133577/2ea5a8ed-f29a-4447-9330-b34c6d52256c)

### CODE REVIEW

**Comment by Angamanga** [comment]
Thank you for your PR and the very detailed description of your work @kivuvarosekivuvan 🙏. @ushahidlee Could you please review?

**Comment by kivuvarosekivuvan** [comment]
@Angamanga Thank you for the compliment. Awaiting for review from @ushahidlee

**Review by Angamanga — CHANGES_REQUESTED** [review]
Thank you for this PR @kivuvarosekivuvan, great work, this issue had many steps to work through. A few comments:
1. I noticed that if I don't make any changes, and press "Cancel", the original text is also wiped from the form. I think you could look at this row: https://github.com/ushahidi/platform-client-mzima/pull/813/files#diff-bcd3dc0cdd5e9cadc9d7cf27b546e46462ffd7d7c62a96f97935e1041c6e06a9L61. That is how the form is filled from the beginning.
2. Great work at creating the confirm-dialog. However, we already have a modal-service that you could use, take a look how it is used when cancelling editing a post here: https://github.com/ushahidi/platform-client-mzima/blob/development/apps/web-mzima-client/src/app/post/post-edit/post-edit.component.ts#L696-L701

Let me know if you have any questions and good work so far!

**Comment by kivuvarosekivuvan** [comment]
@Angamanga Thank you so much for this review :pray: .However, I didn't fully grasp the meaning of the first comment. I would greatly appreciated if you could provide further clarification in order for me to comprehend it accurately and proceed with its execution.

 The second comment is clear, and I will start working on it right away.

**Comment by kivuvarosekivuvan** [comment]
@Angamanga and @ushahidlee I have made the requested changes based on your suggestions. Could you please take a moment to review the updated version and provide me feedback 🙏?

**Comment by Angamanga** [comment]
Hi @kivuvarosekivuvan! Did you figure out my first comment above? I meant that if I press cancel, I would assume the original, saved values should be visible in the form, it should not result in the form being empty 👍

**Comment by kivuvarosekivuvan** [comment]
@Angamanga Thanks🙏. Please clarify whether this is what you mean. I have fixed it

[Screencast from 03-21-2024 05:13:13 PM.webm](https://github.com/ushahidi/platform-client-mzima/assets/124133577/34d9f208-5a8f-4c3c-97ac-9ae4ae79e244)

File: package-lock.json
Comment (Angamanga) [inline package-lock.json:1]: Remove this file from the PR

File: package.json
Comment (Angamanga) [inline package.json:1]: Could you revert back to the old versions here if they are not needed by this PR?

**Review by Angamanga — CHANGES_REQUESTED** [review]
@kivuvarosekivuvan Great work! Nearly there :) Just a few comments in the code.

File: package.json
Comment (kivuvarosekivuvan) [reply package.json:1]: Sure thing! I will do that @Angamanga

File: package-lock.json
Comment (kivuvarosekivuvan) [reply package-lock.json:1]: I will get rid of it here👌

Comment (kivuvarosekivuvan) [reply package-lock.json:1]: @Angamanga Could you please confirm if I did it as expected :pray: , I have made changes.

**Review by Angamanga — APPROVED** [review]
It works now so approving! If you would like, try removing the package-lock entirely from the PR, not from the project itself, so it has the same state as the development-branch. But if its too tricky, I can sort it out before merging :)

**Comment by Angamanga** [comment]
@Angamanga Remove package-lock before merging

**Comment by kivuvarosekivuvan** [comment]
> It works now so approving! If you would like, try removing the package-lock entirely from the PR, not from the project itself, so it has the same state as the development-branch. But if its too tricky, I can sort it out before merging :) 

Hello @Angamanga, I tried to remove it but it was not working as I expected, I can try it again and let you know once I manage it

**Review by Angamanga — APPROVED** [review]

**Comment by Angamanga** [comment]
Merged in this PR: https://github.com/ushahidi/platform-client-mzima/pull/1130

**Pull request closed without merging**
